src/pages/files.tsx

```diff
@@ -1,64 +1,383 @@
+      try {
+        const listing = await props.contentsManager.get('', { content: true });
+        if (listing.type === 'directory' && listing.content) {
+          const files = await Promise.all(
+            (listing.content as Contents.IModel[])
+              .filter(
+                f =>
+                  f.type === 'file' &&
+                  isSupportedFileType({
+                    name: f.name,
+                    type: f.mimetype ?? '',
+                    size: f.size ?? 0,
+                    lastModified: Date.now()
+                  } as File)
+              )
+              .map(async f => {
+                const content = await props.contentsManager.get(f.path, { content: true });
+                const isImage = content.mimetype?.startsWith('image/');
+                const base64Content = isImage
+                  ? (content.content as string)
+                  : btoa(content.content as string);
+
+                return {
+                  id: generateFileId(),
+                  name: f.name,
+                  size: f.size ?? 0,
+                  type: content.mimetype ?? '',
+                  lastModified: Date.now(),
+                  content: `data:${content.mimetype};base64,${base64Content}`
+                };
+              })
+          );
+
+          setUploadedFiles(files);
+        }
+      } catch (err) {
+        console.error('Error loading files:', err); // TODO: Use a better UI alert?
+      }
+    })();
+  }, [props.contentsManager]);
+
+  return (
+    <div className="je-FilesApp">
+      <div className="je-FilesApp-header">
+        <Tile
+          icon={EverywhereIcons.addFile}
+          label="add new"
+          onClick={() => fileUploaderRef.current?.triggerFileSelect()}
+          isLoading={isUploading}
+        />
+      </div>
+      <FileUploader
+        ref={fileUploaderRef}
+        onFilesUploaded={newFiles =>
+          setUploadedFiles(prev => {
+            const filtered = prev.filter(f => !newFiles.some(nf => nf.name === f.name));
+            return [...filtered, ...newFiles];
+          })
+        }
+        contentsManager={props.contentsManager}
+        onUploadStart={() => setIsUploading(true)}
+        onUploadEnd={() => setIsUploading(false)}
+      />
+      <div className="je-FilesApp-content">
+        <div className="je-FilesApp-grid">
+          {uploadedFiles.map(file => (
+            <FileThumbnail
+              key={file.id}
+              file={file}
+              onRemove={id => setUploadedFiles(prev => prev.filter(f => f.id !== id))}
+              contentsManager={props.contentsManager}
+            />
+          ))}
+        </div>
+      </div>
+      <FilesWarningBanner />
+    </div>
+  );
+}
+
 class Files extends ReactWidget {
-  constructor() {
+  private _contentsManager: Contents.IManager;
+  constructor(contentsManager: Contents.IManager) {
     super();
     this.addClass('je-Files');
+    this._contentsManager = contentsManager;
```

Comment:
This is just a stylistic FYI, but you could do this instead:

```suggestion
  constructor(private _contentsManager: Contents.IManager) {
    super();
    this.addClass('je-Files');
```

which saves you two lines of code :)